nilearn/decoding/searchlight.py

```diff
@@ -191,9 +193,21 @@ def _group_iter_search_light(
         if isinstance(cv, KFold):
             kwargs = {"scoring": scoring}
 
-        par_scores[i] = np.mean(
-            cross_val_score(estimator, X[:, row], y, cv=cv, n_jobs=1, **kwargs)
-        )
+        if y is None:
+            y_dummy = np.array(
+                [0] * (X.shape[0] // 2) + [1] * (X.shape[0] // 2)
+            )
+            estimator.fit(
+                X[:, row], y_dummy[: X.shape[0]]
+            )  # Ensure the size matches X
+            par_scores[i] = np.mean(estimator.decision_function(X[:, row]))
```

Comment:
```suggestion
        if y is None:
            y_dummy = np.array(
                [0] * (X.shape[0] // 2) + [1] * (X.shape[0] // 2)
            )
            estimator.fit(
                X[:, row], y_dummy[: X.shape[0]]
            )  # Ensure the size matches X
            par_scores[i] = np.mean(estimator.decision_function(X[:, row]))
```

I am not sure we want to handle `y is None` cases:

- I do not think that they are likely to happen unless the user makes an error
- I think that they should throw an error and we should probably let the estimator deal with this

Maybe I am missing something: What was your reason for adding this?

Reply:
OK. Read your comment in the transform method.

I think that we should probably add a comment here to explain why we are handling `y = None`, probably by updating the doc string of _group_iter_search_light